### PR TITLE
Added Travis YAML file to support automated unit testing

### PR DESCRIPTION
**Pull request by MichaelCampbell**

### CODE REVIEW

**Comment by htinlinn** [comment]
`mParticleTests` don't seem to be running on Travis; only `SimcoeTests` are being run...

**Comment by yoyoyoseob** [comment]
How would I go about testing this?

**Comment by MichaelCampbell** [comment]
@yoyoyoseob you can go to the `SimcoeTests` scheme and make sure the unit tests are running for both Simcoe and mParticle.  You can also try running `xcodebuild test -project Simcoe.xcodeproj -scheme SimcoeTests -destination 'platform=iOS Simulator,name=iPhone 6,OS=10.0' | xcpretty` and see if that works locally.

It's still failing for me on Travis, so I don't think this is fully baked as yet.

File: .travis.yml [hunk 1]
```diff
@@ -0,0 +1,18 @@
...
+cache: cocoapods
+podfile: Podfile
+before_install:
+- rvm install $(cat .ruby-version) 
+- rvm use $(cat .ruby-version)
+- gem install bundler
+- bundle install
+- pod install
+script:
+- set -o pipefail && xcodebuild test -project Simcoe.xcodeproj -scheme SimcoeTests -destination 'platform=iOS Simulator,name=iPhone 7,OS=10.0' | xcpretty
```
Comment (htinlinn) [inline .travis.yml]: Changing `-project Simcoe.xcodeproj` to `-workspace Simcoe.xcworkspace` seems to fix the issues on my local computer...

**Review by htinlinn — CHANGES_REQUESTED** [review]

**Review by htinlinn — APPROVED** [review]

**Review by htinlinn — APPROVED** [review]

**Pull request merged**